packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -416,3 +440,88 @@ const toChatError = (target: ChatErrorTarget, error: unknown): ChatError => {
   }
   return new ChatError(target, new Error(`${error}`));
 };
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+class ResourceDownloadQueue {
+  private _messageQueue: ChatMessageWithStatus[] = [];
+  private _context: ChatContext;
+  private _credential: CommunicationTokenCredential;
+  private _isRequesting = false;
+
+  constructor(context: ChatContext, credential: CommunicationTokenCredential) {
+    this._context = context;
+    this._credential = credential;
+  }
+
+  public isRequesting(): boolean {
+    return this._isRequesting;
+  }
+
+  public containsMessage(message: ChatMessageWithStatus): boolean {
```

Comment:
Do we need this to public? Or should this only be called from the addMessage function?

Reply:
I think it should be public. In the case of a queue, it shouldn't manage whether or not it has duplicates. The caller should be handle that business logic.

Reply:
It would be simpler for the caller to not need to know about whether a particular message has been cached already.